src/main/src/main/java/org/geoserver/catalog/impl/ModificationProxyCloner.java

```diff
@@ -194,7 +195,13 @@ public int compare(Class c1, Class c2) {
             // nothing to copy
             return null;
         }
-        Collection<T> copy = source.getClass().newInstance();
+        Collection<T> copy;
+        try {
+            copy = source.getClass().newInstance();
+        } catch (InstantiationException e) {
+            //we'll just pick something
+            copy = new ArrayList<T>();
```

Comment:
You could switch the class to static and pass around the fields it needs, then give it also a empty constructor, that would fix issues without assuming a non cloneable collection is necessarily a list.